WordPress/src/main/java/org/wordpress/android/WordPressDB.java

```diff
@@ -1010,37 +1011,70 @@ public void savePosts(List<?> postsList, int localBlogId, boolean isPage, boolea
         }
     }
 
-    public List<PostsListPost> getPostsListPosts(int blogId, boolean loadPages) {
-        List<PostsListPost> posts = new ArrayList<PostsListPost>();
-        Cursor c;
-        c = db.query(POSTS_TABLE,
-                new String[] { "id", "blogID", "title",
-                        "date_created_gmt", "post_status", "isUploading", "localDraft", "isLocalChange" },
-                "blogID=? AND isPage=? AND NOT (localDraft=1 AND uploaded=1)",
-                new String[] {String.valueOf(blogId), (loadPages) ? "1" : "0"}, null, null, "localDraft DESC, date_created_gmt DESC");
-        int numRows = c.getCount();
-        c.moveToFirst();
+    /*
+     * returns list of posts for use in the post list fragment
+     */
+    public PostsListPostList getPostsListPosts(int localBlogId, boolean loadPages) {
+        PostsListPostList listPosts = new PostsListPostList();
 
-        for (int i = 0; i < numRows; ++i) {
-            String postTitle = StringUtils.unescapeHTML(c.getString(c.getColumnIndex("title")));
-
-            // Create the PostsListPost and add it to the Array
-            PostsListPost post = new PostsListPost(
-                    c.getInt(c.getColumnIndex("id")),
-                    c.getInt(c.getColumnIndex("blogID")),
-                    postTitle,
-                    c.getLong(c.getColumnIndex("date_created_gmt")),
-                    c.getString(c.getColumnIndex("post_status")),
-                    SqlUtils.sqlToBool(c.getInt(c.getColumnIndex("localDraft"))),
-                    SqlUtils.sqlToBool(c.getInt(c.getColumnIndex("isLocalChange"))),
-                    SqlUtils.sqlToBool(c.getInt(c.getColumnIndex("isUploading")))
-            );
-            posts.add(i, post);
-            c.moveToNext();
+        String[] args = {Integer.toString(localBlogId), Integer.toString(loadPages ? 1 : 0)};
+        String query = "blogID=? AND isPage=? AND NOT (localDraft=1 AND uploaded=1)";
+
+        Cursor c = db.query(POSTS_TABLE, null, query, args, null, null, "localDraft DESC, date_created_gmt DESC");
+        try {
+            while (c.moveToNext()) {
+                listPosts.add(new PostsListPost(getPostFromCursor(c)));
+            }
+            return listPosts;
+        } finally {
+            SqlUtils.closeCursor(c);
+        }
+    }
+
+    private Post getPostFromCursor(Cursor c) {
+        Post post = new Post();
+
+        post.setLocalTableBlogId(c.getInt(c.getColumnIndex("blogID")));
```

Comment:
It'd be nice to have these column names defined somewhere (like [this](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/WordPressDB.java#L56)). Probably best for another PR though and not really in scope here since you're just moving these lines.